### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by shahbuland**
- Add docstrings to any major objects missing them
- Add docstrings to configs and data classes
- Add RTD style documentation

### CODE REVIEW

File: docs/source/configs.rst [hunk 1]
```diff
@@ -21,16 +21,14 @@ the specific method being used (i.e. ILQL or PPO)
     :members:
 
 .. autoclass:: trlx.data.method_configs.MethodConfig
-    :members:
+    :members:  
 
-**PPO**
+**PPO**
```
Comment (thedch) [inline docs/source/configs.rst:26]: Any reason to add trailing whitespace?

File: trlx/data/ppo_types.py [hunk 2]
```diff
@@ -23,13 +21,11 @@ class PPORLElement:
...
     :type rewards: torch.Tensor
     """
-
-    query_tensor: TensorType["query_size"]
-    response_tensor: TensorType["response_size"]
-    logprobs: TensorType["response_size", "vocab_size"]
-    values: TensorType["response_size"]
-    rewards: TensorType["response_size"]
-
+    query_tensor : TensorType["query_size"]
```
Comment (thedch) [inline trlx/data/ppo_types.py:24]: Why add the space before and after the colon? Does this pass black?

**Comment by shahbuland** [comment]
@thedch Bit confused by results of black. Fixed the simple things, but code-quality test still failing. If you have time, could you take a look and give some pointers on where things are going wrong?

File: README.md [hunk 3]
```diff
@@ -3,6 +3,10 @@
 
 # Transformer Reinforcement Learning X
 
+[![Docs Status][docs-image]][docs-url]
+
+**[Documentation](https://trlX.readthedocs.io)**
```
Comment (reciprocated) [inline README.md:8]: there are 4 links to the documentation, which are redundant?

File: trlx/data/configs.py [hunk 4]
```diff
@@ -63,7 +63,7 @@ class TrainConfig:
     :param learning_rate_target: Target learning rate after decay
     :type learning_rate_target: float
 
-    :param checkpoint_interval: Save model every checkpoint_interval steps
+    :param checkpoint_interval: Save model every save_interval steps
```
Comment (reciprocated) [inline trlx/data/configs.py:66]: checkpoint_interval here

**Review by reciprocated — CHANGES_REQUESTED** [review]
lgtm

**Comment by LouisCastricato** [comment]
@shahbuland answer the comments and lets merge.

File: trlx/data/configs.py (hunk 4 above)
Comment (shahbuland) [reply trlx/data/configs.py]: Done, for this and  readme change

**Review by LouisCastricato — APPROVED** [review]

**Comment by LouisCastricato** [comment]
Will merge after check passes.

**Pull request merged**